### PR TITLE
refactor/args: make args part of config module

### PR DESCRIPTION
**Pull request by drahnr**
## What does this PR accomplish?

 * 🪣 Misc

Refactor args usage, improve unit test coverage; add tests for the interplay of `Config` and `Args` by means of introducing `enum UnifiedArgs`.

## 📜 Checklist

 * [x] Works on the `./demo` sub directory
 * [x] Test coverage is excellent and passes
 * [x] Documentation is thorough

### CODE REVIEW

File: src/config/hunspell.rs [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+    /// A regular expression, whose capture groups will be checked, instead of the initial token.
+    /// Only the first one that matches will be used to split the word.
+    #[serde(default)]
+    pub transform_regex: Vec<WrappedRegex>,
+    /// Allow concatenated words instead of dashed connection.
+    /// Note that this only applies, if one of the suggested replacements has an item that is
+    /// equivalent except for addition dashes (`-`).
+    #[serde(default)]
+    pub allow_concatenation: bool,
+    /// The counterpart of `allow_concatenation`. Accepts words which have repalcement suggestions
```
Comment (KuabeM) [inline src/config/hunspell.rs:21]:
```suggestion
    /// The counterpart of `allow_concatenation`. Accepts words which have replacement suggestions
```

File: src/config/hunspell.rs [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+    /// that contain additional dashes.
+    #[serde(default)]
+    pub allow_dashes: bool,
+}
+
+impl Default for Quirks {
+    fn default() -> Self {
+        // use some for default, so for generating the default config has the default values
```
Comment (KuabeM) [inline src/config/hunspell.rs:29]: I think that comment refers to the old version right?

File: src/config/hunspell.rs [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+}
+
+#[derive(Deserialize, Serialize, Debug, Clone)]
+#[serde(deny_unknown_fields)]
+pub struct HunspellConfig {
+    /// The language we want to check against, used as the dictionary and affixes file name.
+    // TODO impl a custom xx_YY code deserializer based on iso crates
+    pub lang: Option<String>,
+    /// Additional search dirs for `.dic` and `.aff` files.
+    // must be option so it can be omitted in the config
```
Comment (KuabeM) [inline src/config/hunspell.rs:60]: Also no longer an option:
```suggestion
```

File: src/config/args.rs [hunk 4]
```diff
@@ -0,0 +1,505 @@
...
+            #[cfg(feature = "hunspell")]
+            if !checkers.contains(&CheckerType::Hunspell) {
+                if !config.hunspell.take().is_some() {
+                    warn!("Hunspell was never configured.")
+                }
+            }
+            #[cfg(feature = "nlprule")]
+            if !checkers.contains(&CheckerType::NlpRules) {
+                if !config.languagetool.take().is_some() {
+                    warn!("Nlprules checker was never configured.")
```
Comment (KuabeM) [inline src/config/args.rs:191]:
```suggestion
                if !config.nlprules.take().is_some() {
                    warn!("Nlprules checker was never configured.")
```
?

File: src/config/args.rs [hunk 5]
```diff
@@ -0,0 +1,505 @@
...
+            if checkers.iter().unique().count() == EXPECTED_COUNT {
+                bail!("Argument override for checkers disabled all checkers")
+            }
+        }
+        Ok(())
+    }
+
+    /// Load configuration with fallbacks.
+    ///
+    /// When explicitly
```
Comment (KuabeM) [inline src/config/args.rs:219]: I think something's missing here

File: src/config/hunspell.rs (hunk 1 above)
Comment (drahnr) [reply src/config/hunspell.rs]: I just ran `cargo spellcheck --fix` on the codebase, and it turns out CI should enforce it. There were quite a few.

File: src/config/args.rs (hunk 4 above)
Comment (drahnr) [reply src/config/args.rs]: Very good catch!

**Pull request merged**